src/client/hud.cpp

```diff
@@ -378,19 +380,23 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 				if (e->offset.X < -20)
 					textsize = font_scaled->getDimension(text.c_str());
 #endif
-				v2s32 offset((e->align.X - 1.0) * (textsize.Width / 2),
-				             (e->align.Y - 1.0) * (textsize.Height / 2));
+				v2s32 offset(0, (e->align.Y - 1.0) * (textsize.Height / 2));
 				core::rect<s32> size(0, 0, e->scale.X * m_scale_factor,
 				                     text_height * e->scale.Y * m_scale_factor);
 				v2s32 offs(e->offset.X * m_scale_factor,
 				           e->offset.Y * m_scale_factor);
 #ifdef __ANDROID__
 				if (e->offset.X < -20)
-					font_scaled->draw(text.c_str(), size + pos + offset + offs, color);
-				else
+					textfont = font_scaled;
 #endif
+				auto wss = std::wstringstream(text);
+				std::wstring line;
+			    while (std::getline(wss, line, L'\n'))
 				{
-					textfont->draw(text.c_str(), size + pos + offset + offs, color);
+					core::dimension2d<u32> linesize = textfont->getDimension(line.c_str());
+					v2s32 line_offset((e->align.X - 1.0) * (linesize.Width / 2), 0);
```

Comment:
Should probably be negated so that `1` means right justified and `-1` means left justified.